src/f32ext.rs

```diff
@@ -1,22 +1,103 @@
 //! Floating point operations which would ordinarily be provided by `std`
 //! but aren't available in `no_std`.
 
-use core::mem;
+#![allow(clippy::unreadable_literal)]
+
+use core::{f32::consts::PI, mem};
 
 /// `std`-like extension methods for math support
 pub(crate) trait F32Ext: Sized {
+    /// Compute absolute value
+    fn abs(self) -> f32;
+
+    /// Compute four quadrant arctangent normalized between `[0, 2)`
+    fn atan2_norm(self, other: f32) -> f32;
+
+    /// Compute four quadrant arctangent
+    fn atan2(self, other: f32) -> f32;
+
+    /// Compute radians for a number between `0..n`
+    ///
+    /// Panics if `self` > `n`
+    fn radians_max<N: Into<f32>>(self, n: N) -> f32;
+
+    /// Assuming self is normalized between `[0, 2)`, compute radians
+    fn radians_norm(self) -> f32;
+
+    /// Compute square root
     fn sqrt(self) -> Self;
 }
 
 impl F32Ext for f32 {
-    /// Square root approximation
+    fn abs(self) -> f32 {
+        if self < 0.0 {
+            -self
+        } else {
+            self
+        }
+    }
+
+    fn atan2_norm(self, other: f32) -> f32 {
+        atan2_norm_approx(self, other)
+    }
+
+    fn atan2(self, other: f32) -> f32 {
+        atan2_approx(self, other)
+    }
+
+    fn radians_max<N>(self, n: N) -> f32
+    where
+        N: Into<f32>,
+    {
+        let n_normalized = self / n.into();
```

Comment:
What if `n = 0.0`?

Reply:
Note above this is an internal i.e. `pub(crate)` trait, and so far the only values passed for `n` are nonzero constants, e.g.:

https://github.com/NeoBirth/accelerometer.rs/pull/14/files#diff-6ca27e5d8d11516966e24e34a4785397R21